### PR TITLE
Updated Cloud Guide navigation for accessible Docker launch instructions

### PR DESCRIPTION
**Pull request by meker12**
## Purpose of this pull request

Make Docker Launch instructions more accessible:
- Added the Docker Development > Launch Docker navigation heading
- Changes the top-level navigation title from "Launch Docker" to
"Configure and launch Docker"
- Updated the Launch developer and Launch production topics to include
 links to the Docker configuration topics
- Changed title of "Launch modes" subsection to add what launch modes
are for == "Launch modes for configuring the Docker environment"

## Affected DevDocs pages
- https://devdocs.magento.com/cloud/docker/docker-config.html
- https://devdocs.magento.com/cloud/docker/docker-development.html
- https://devdocs.magento.com/cloud/docker/docker-mode-production.html
- https://devdocs.magento.com/cloud/docker/docker-quick-reference.html
- https://devdocs.magento.com/cloud/reference/ece-tools-reference.html
- https://devdocs.magento.com/cloud/release-notes/cloud-release-archive.html

### CODE REVIEW

File: src/cloud/docker/docker-quick-reference.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ functional_areas:
   - Docker
 ---
 
+{:.bs-callout-info}
+See [Configure Docker] for details on configuring and launching a {{site.data.var.mcd-prod}} environment.
```
Comment (hguthrie) [inline src/cloud/docker/docker-quick-reference.md]: Why not use the include note you created? It sends you to the same place.

File: src/cloud/docker/docker-quick-reference.md [hunk 2]
```diff
@@ -39,8 +42,8 @@ docker-compose -f docker-compose.yml -f docker-compose-custom.yml [-f more-custo
 
 | Option       | Key              | Available values
 | ------------ | ---------------- | ------------------
-| [Mode]({{site.baseurl}}/cloud/docker/docker-config.html#launch-modes)         | `--mode`, `-m`   | production, developer
-| [File synchronization engine]({{site.baseurl}}/cloud/docker/docker-config.html#launch-modes) | `--sync-engine` | native (default), docker-sync, mutagen
+| [Mode]({{site.baseurl}}/cloud/docker/docker-config.html#launch-modes-for-configuring-the-docker-environment)         | `--mode`, `-m`   | production, developer
```
Comment (hguthrie) [inline src/cloud/docker/docker-quick-reference.md]: yuck

File: src/cloud/docker/docker-quick-reference.md [hunk 3]
```diff
@@ -39,8 +42,8 @@ docker-compose -f docker-compose.yml -f docker-compose-custom.yml [-f more-custo
...
+| [File synchronization engine]({{site.baseurl}}/cloud/docker/docker-config.html#launch-modes-for-configuring-the-docker-environment) | `--sync-engine` | native (default), docker-sync, mutagen
```
Comment (hguthrie) [inline src/cloud/docker/docker-quick-reference.md]: There might be a more specific place to link that is related to the syncing. 
https://devdocs.magento.com/cloud/docker/docker-syncing-data.html

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Pull request merged**